coalib/bearlib/languages/documentation/DocumentationExtraction.py

```diff
@@ -250,4 +251,58 @@ def extract_documentation_with_markers(content, docstyle_definition):
             marker_dict,
             docstyle_definition)
         if doc:
+            # Instantiate padding
+            top_padding = 0
+            bottom_padding = 0
+            start_index = doc.range.start.line - 2
+            end_index = doc.range.end.line
+            while start_index >= 0 and content[start_index] == '\n':
+                top_padding += 1
+                start_index -= 1
+            # If the end_index is instantiated above the len(content) i.e.
+            # In case where ending marker of docstring is at the last line.
+            # Then the bottom_padding will be default to 0. This will also
+            # prevent IndexError raised by content[end_index].
+            while end_index < len(content) and content[end_index] == '\n':
+                # This condition will take place if theres an inline docstring
+                # following documentation.
+                if ((doc.marker[2]+'\n') != content[end_index-1][-4:]
+                        and bottom_padding == 0):
+                    break
+                bottom_padding += 1
+                end_index += 1
+            doc.top_padding = top_padding
+            doc.bottom_padding = bottom_padding
+            doc.range = TextRange.from_values(
+                start_index + 2,
+                1 if top_padding > 0 else doc.range.start.column,
+                end_index,
+                1 if bottom_padding > 0 else doc.range.end.column)
+
+            # Instantiate docstring_type
+            class_regex = re.compile(
+                doc.docstyle_definition.docstring_type_regex.class_sign)
+            function_regex = re.compile(
+                doc.docstyle_definition.docstring_type_regex.func_sign)
+
+            # Reinitializing start_index and end_index as `range` is changed.
+            start_index = doc.range.start.line - 2
+
+            # End line differs when mid marker and end marker is different
+            if doc.marker[1] == doc.marker[2]:
+                end_index = doc.range.end.line - 1
+            else:
+                end_index = doc.range.end.line
+
+            # We check for the class regex and function regex
+            # before and after the documentation.
```

Comment:
introducing `docstring_type_position` is the correct solution. The other will not work, you'll only make it work for Python. Maybe `docstring_position` should suffice, I doubt that styles will position the docstring on different sides for different types.